libraries/AP_HAL_ChibiOS/UARTDriver.cpp

```diff
@@ -686,6 +688,14 @@ ssize_t UARTDriver::read(uint8_t *buffer, uint16_t count)
         update_rts_line();
     }
 
+#if HAL_LOG_UART_STATS
+    {
+        WITH_SEMAPHORE(_stats.sem);
+        _stats.rx_bytes += count;
+        update_readbuf_stats();
```

Comment:
Why not have `update_readbuf_stats` take the number of new read bytes as an argument and adding it internally? It seems like you always basically have a new number to add (or 0 I guess for the clear case)